### DISCUSSION TITLE
BlockManager.getConfig.getSorter() not working

### DISCUSSION THREAD

**Discussion by esha-fatima** [GrapesJS/grapesjs#5313] [Q&A]
Hello, what could be an alternative to achieve this? getSorter has been removed from the config object. @artf

**Comment by lexoyo** [comment]
Not sure what you are trying to achieve? 

This is how I use the sorter to get drop targets
https://github.com/silexlabs/grapesjs-symbols/blob/25f70828c12a094561f494b00d44a9e13c318c73/src/view/SymbolsView.ts#L97

**Reply by esha-fatima** [reply]
I am trying to ` const sorter = editor.BlockManager.getConfig().getSorter()` get the sorter and then check if it is a valid insert point for a new block using `valid = sorter.validTarget(contentContainer.getEl(), content);`
i am considering all of these checks in a function that is attached as a callback to appendOnClick -
 blockManager: {
        appendTo: //',
        appendOnClick: appendBlockOnClick,
        blocks: //,
    },

**Reply by lexoyo** [reply]
I'm not sure you should do that in any case as the sorter is an internal api right?

Not sure really :))

If you want to give context here I may have ideas. Or we wait for artf opinion

**Reply by artf** [reply]
Yeah, you can rely on `editor.Components.canMove(target, source, index?)` for that check. 
It's not yet "public"/documented method (maybe in the next release) but you can make use of it.
The `source` could be a component, component definition (eg. `block.content` usually defines the component definition), or an HTML string
